wandb-scim/examples/custom_roles_example.py

```diff
@@ -0,0 +1,157 @@
+    """
+    try:
+        # Create a new custom role
+        create_role_response = custom_role.create(
+            request_payload={
+                "permissionJson": permission_json,
+                "inheritedFrom": inherited_from
+            }
+        )
+        print(create_role_response)
+    except requests.exceptions.RequestException as e:
+        print(f"Error occurred during API request: {str(e)}")
+
+def get_custom_role(custom_role, role_id):
+    """
+    Retrieves details of a specific custom role.
+
+    Args:
+        custom_role (CustomRole): An instance of the CustomRole class.
+        role_id (str): The ID of the custom role to retrieve.
+    """
+    try:
+        # Get details of a custom role
+        get_role_response = custom_role.get(role_id)
+        print(get_role_response)
+    except requests.exceptions.RequestException as e:
+        print(f"Error occurred during API request: {str(e)}")
+
+def get_all_roles(custom_role):
+    """
+    Retrieves details of all custom roles.
+
+    Args:
+        custom_role (CustomRole): An instance of the CustomRole class.
+    """
+    try:
+        # Get all custom roles
+        get_all_roles_response = custom_role.get_all()
+        print(get_all_roles_response)
+    except requests.exceptions.RequestException as e:
+        print(f"Error occurred during API request: {str(e)}")
+
+def add_permission(custom_role, role_id, permission_json):
+    """
+    Adds permission to a custom role.
+
+    Args:
+        custom_role (CustomRole): An instance of the CustomRole class.
+        role_id (str): The ID of the custom role to update.
+        permission_json (str): JSON string representing the permission to add.
+    """
+    try:
+        # Add permission to a custom role
+        update_role_response = custom_role.add_permissions(
+            role_id,
+            request_payload={
+                "permissionJson": permission_json
+            }
+        )
+        print(update_role_response)
+    except requests.exceptions.RequestException as e:
+        print(f"Error occurred during API request: {str(e)}")
+
+def remove_permission(custom_role, role_id, permission_json):
+    """
+    Removes permission from a custom role.
+
+    Args:
+        custom_role (CustomRole): An instance of the CustomRole class.
+        role_id (str): The ID of the custom role to update.
+        permission_json (str): JSON string representing the permission to remove.
+    """
+    try:
+        # Remove permission from a custom role
+        remove_role_response = custom_role.remove_permission(
+            role_id,
+            request_payload={
+                "permissionJson": permission_json
+            }
+        )
+        print(remove_role_response)
+    except requests.exceptions.RequestException as e:
+        print(f"Error occurred during API request: {str(e)}")
+
+def update_custom_role(custom_role, role_id, role_name, role_description, inherited_from):
```

Comment:
Repeating the feedback here - we would like to update this conditionally for every field based on if it's provided in the input request. All three fields - name, description and inheritedFrom are optional here, so one could specify just one of those, or two of those, or all three to be updated. So we should expect either of those fields to be None and construct the SCIM request field by field only if it's available in the payload.